procedures/statement_performance_analyzer.sql

```diff
@@ -0,0 +1,792 @@
+                SET @sys.tmp.SQL = CONCAT(@sys.tmp.SQL, ' LIMIT ', @sys.statement_analyzer_limit);
+            END IF;
+            IF (@sys.debug = 'ON') THEN
+                SELECT @sys.tmp.SQL AS 'Debug';
+            END IF;
+            PREPARE stmt_select_view FROM @sys.tmp.SQL;
+            EXECUTE stmt_select_view;
+            DEALLOCATE PREPARE stmt_select_view;
+        END IF;
+
+        IF (FIND_IN_SET('with_temp_tables', in_views)) THEN
+            SELECT CONCAT('Top ', @sys.statement_analyzer_limit, ' Queries with Internal Temporary Tables') AS 'Next Output';
+            SET @sys.tmp.SQL = REPLACE(
+                                  (SELECT VIEW_DEFINITION
+                                     FROM information_schema.VIEWS
+                                    WHERE TABLE_SCHEMA = DATABASE() AND TABLE_NAME = 'statements_with_temp_tables'
+                                  ),
+                                  '`performance_schema`.`events_statements_summary_by_digest`',
+                                  v_digests_table
+                               );
+            IF (@sys.statement_analyzer_limit > 0) THEN
+                SET @sys.tmp.SQL = CONCAT(@sys.tmp.SQL, ' LIMIT ', @sys.statement_analyzer_limit);
+            END IF;
+            IF (@sys.debug = 'ON') THEN
+                SELECT @sys.tmp.SQL AS 'Debug';
+            END IF;
+            PREPARE stmt_select_view FROM @sys.tmp.SQL;
+            EXECUTE stmt_select_view;
+            DEALLOCATE PREPARE stmt_select_view;
+        END IF;
+
+        IF (FIND_IN_SET('custom', in_views)) THEN
+            SELECT CONCAT('Top ', @sys.statement_analyzer_limit, ' Queries Using Custom View') AS 'Next Output';
+
+            IF (@sys.statement_analyzer_view IS NULL) THEN
+                SET @sys.statement_analyzer_view = sys.sys_get_config('sys.statement_analyzer_view', NULL);
+            END IF;
+            IF (@sys.statement_analyzer_view IS NULL) THEN
+                SIGNAL SQLSTATE '45000'
+                   SET MESSAGE_TEXT = 'The @sys.statement_analyzer_view user variable must be set with the view or query  to use';
+            END IF;
+
+            IF (NOT INSTR(@sys.statement_analyzer_view, ' ')) THEN
+                -- No spaces, so can't be a query
+                IF (NOT INSTR(@sys.statement_analyzer_view, '.')) THEN
+                    -- No . in the table name - use current database
+                    -- DATABASE() will be the database of the procedure
+                    SET v_custom_db   = DATABASE(),
+                        v_custom_name = @sys.statement_analyzer_view;
+                ELSE
+                    SET v_custom_db   = SUBSTRING_INDEX(@sys.statement_analyzer_view, '.', 1);
+                    SET v_custom_name = SUBSTRING(@sys.statement_analyzer_view, CHAR_LENGTH(v_custom_db)+2);
+                END IF;
+
+                CALL table_exists(v_custom_db, v_custom_name, v_custom_view_exists);
```

Comment:
Prefix this with "sys.table_exists", these can be called from outside of the sys schema.